File: src/main/java/mods/eln/sim/electrical/mna/component/Monopole.java
package mods.eln.sim.electrical.mna.component;

import mods.eln.sim.electrical.mna.state.State;
import mods.eln.sim.electrical.mna.state.VoltageState;

public abstract class Monopole extends Component {

    VoltageState pin;

    public Monopole connectTo(VoltageState pin) {
        this.pin = pin;
        if (pin != null) pin.add(this);
        return this;
    }

    @Override
    public State[] getConnectedStates() {
        return new State[]{pin};
    }
}

File: src/main/java/mods/eln/sim/mna/component/Monopole.kt
package mods.eln.sim.mna.component

import mods.eln.sim.mna.state.State
import mods.eln.sim.mna.state.VoltageState

abstract class Monopole : Component() {

    internal var pin: VoltageState? = null

    override fun getConnectedStates() = arrayOf<State?>(pin)

    fun connectTo(pin: VoltageState?): Monopole {
        this.pin = pin
        pin?.add(this)
        return this
    }
}
